src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+      format.println("Make_Plan_GPU: Processor kind is {}", c.legion_processor_kind(proc))
+
+      if c.legion_processor_kind(proc) == c.TOC_PROC then
+        format.println("Processor is TOC, so running GPU functions")
+        var i = c.legion_processor_address_space(proc)
+        regentlib.assert(address_space == i, "make_plan_gpu_batch must be executed on a processor in the same address space")
+
+        --Get input and output bases
+        var input_base = get_base_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+        var output_base = get_base_out(rect_t_out(output.ispace.bounds), __physical(output)[0], __fields(output)[0])
+        var lo = input.ispace.bounds.lo:to_point()
+        var hi = input.ispace.bounds.hi:to_point()
+        var n : int[dim] --n is an array of size dim with the size of each dimension in the entries
+        ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+
 
+        --Define 'n' array: n is an array of size rank, describing the size of each dimension. For batched transforms, we want to exclude the last dimension as that is the number of batches
+        --AK Note: There must be a better way to copy/slice arrays in regent instead of naively using this for loop...
+        var n_batch : int[dim-1]
+        for i = 0, dim do
+          n_batch[i] = n[i]
+        end
+
+         --Set idist: idist is the distance between the first element of two consecutive batches
+        --In a transform where each batch is a 256x256 complex64 transform, offset_1 will be 16, offset_2 will be 16*256, and offet_3 willbe 16*256*256
+        --idist should be 256*256 in this case, so we want offset_3/offset_1
+        var offset_in = get_offset_in(rect_t_in(input.ispace.bounds), __physical(input)[0], __fields(input)[0])
+        var offset_1 = offset_in[0].offset
+        var offset_2 = offset_in[1].offset
+        var offset_3 = offset_in[2].offset
+        var i_dist = offset_3/offset_1
+
+        format.println("n[0] = {}, n[1] = {}, n[2] = {}, n_batch[0] = {}, n_batch[1] = {}, i_dist = {}", n[0], n[1], n[2], n_batch[0], n_batch[1], i_dist)
+
+        --Call cufftPlanMany: cufftResult cufftPlanMany(cufftHandle *plan, int rank, int *n, int *inembed, int istride, int idist, int *onembed, int ostride, int odist, cufftType type, int batch) --rank = dimensionality of transform (1,2,3)
+        
+        --Taking a transform where {256x256x7} is passed (i.e. 7 batches of 256x256), the parameters should be as follows:
+        --rank[In] – How many dimensions a single transform has: 2
+        --n[In] - Array of size rank, describing the size of each dimension: {256, 256}
+        --inembed[In] – Array of size rank. Here we pass the real dimensions of the input array. It should contain the dimension size + padding: {256, 256} again, because input is not padded.
+        --istride[In] – Stride between two consecutive elements in lowest dimension: 1
```

Comment:
Stride isn't actually guaranteed to be 1, is it?

Reply:
Still looking into this

Reply:
From https://docs.nvidia.com/cuda/cufft/index.html: The istride and ostride parameters denote the distance between two successive input and output elements in the least significant (that is, the innermost) dimension respectively. In a single 1D transform, if every input element is to be used in the transform, istride should be set to 
; if every other input element is to be used in the transform, then istride should be set to 
. Similarly, in a single 1D transform, if it is desired to output final elements one after another compactly, ostride should be set to 
; if spacing is desired between the least significant dimension output data, ostride should be set to the distance between the elements.

I think we always want the input/output to be 'compact'?